components/ai_chat/resources/untrusted_conversation_frame/components/assistant_task/use_extract_task_data.ts

```diff
@@ -0,0 +1,71 @@
+// Copyright (c) 2025 The Brave Authors. All rights reserved.
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this file,
+// You can obtain one at https://mozilla.org/MPL/2.0/.
+
+import * as React from 'react'
+import * as Mojom from '../../../common/mojom'
+
+// TODO(https://github.com/brave/brave-browser/issues/48535):
+// Should be Navigate and TODO tools.
+const importantToolNames = [Mojom.NAVIGATE_TOOL_NAME]
+
+export default function useExtractTaskData(
+  assistantEntryGroup: Mojom.ConversationTurn[],
+) {
+  return React.useMemo(() => {
+    // Individual tasks, split by CompletionEvent
+    let taskItems: Mojom.ConversationEntryEvent[][] = [[]]
+    // All completion events are allowed the links provided by the whole response
+    // group.
+    const allowedLinks: string[] = []
+    // Most recent of each type of important tool use event
+    const importantToolUseEvents: Partial<
+      Record<(typeof importantToolNames)[number], Mojom.ToolUseEvent>
+    > = {}
+
+    for (const event of assistantEntryGroup.flatMap(
+      (entry) => entry.events ?? [],
+    )) {
+      if (!event) {
+        continue
+      }
+      if (event.completionEvent) {
+        // Start a new task item for every completion event
+        taskItems.push([event])
```

Comment:
Hey sorry, does this mean completion events will be grouped with the next task?

`[1, 2, 'complete', 3, 4, 'complete']`
will result in
```
[1, 2]
['complete', 3, 4]
['complete']
```

Is that intentional? I (possibly naively) thought the completion should be for the tasks before:
```
[1, 2, 'complete']
[3, 4, 'complete']
```

I guess the above would be working if we were iterating in reverse order?

Reply:
That said, I'm pretty sure I'm just misunderstanding `completetionEvent`